File: pathfinder-api/src/main/java/de/cubbossa/pathfinder/storage/NodeStorageImplementation.java
package de.cubbossa.pathfinder.storage;

import de.cubbossa.pathfinder.node.Node;
import java.util.Collection;
import java.util.Optional;
import java.util.UUID;

public interface NodeStorageImplementation<N extends Node> {

  Optional<N> loadNode(UUID uuid);

  Collection<N> loadNodes(Collection<UUID> ids);

  Collection<N> loadAllNodes();

  void saveNode(N node);

  void deleteNodes(Collection<N> node);
}

File: pathfinder-api/src/main/java/de/cubbossa/pathfinder/storage/NodeStorageImplementation.kt
package de.cubbossa.pathfinder.storage

import de.cubbossa.pathfinder.node.Node
import java.util.*

interface NodeStorageImplementation<N : Node> {

    fun loadNode(uuid: UUID): N?

    fun loadNodes(ids: Collection<UUID>): Collection<N>

    fun loadAllNodes(): Collection<N>

    fun saveNode(node: N)

    fun deleteNodes(node: Collection<N>)
}
